### PR TITLE
Kubeadm config reference

### PR DESCRIPTION
**Pull request by tengqm**
Instead of redirecting users to the godoc site for kubeadm configuration options, we'd better point them to the generated reference.

### CODE REVIEW

**Comment by tengqm** [comment]
/hold
Waiting for the config reference PR to be merged.

**Comment by jihoon-seo** [comment]
The config reference PR: #28871

File: content/en/docs/setup/production-environment/tools/kubeadm/dual-stack-support.md [hunk 1]
```diff
@@ -150,3 +157,5 @@ networking:
 
 * [Validate IPv4/IPv6 dual-stack](/docs/tasks/network/validate-dual-stack) networking
 * Read about [Dual-stack](/docs/concepts/services-networking/dual-stack/) cluster networking
+* Learn more about the kubeadm [configuration file](/docs/reference/config-api/kubeadm-config.v1beta2/)
```
Comment (sftim) [inline content/en/docs/setup/production-environment/tools/kubeadm/dual-stack-support.md:160]:
```suggestion
* Learn more about the kubeadm [configuration format](/docs/reference/config-api/kubeadm-config.v1beta2/)
```

File: content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-certs.md [hunk 2]
```diff
@@ -161,7 +161,7 @@ The built-in signer is part of [`kube-controller-manager`](/docs/reference/comma
 
 To activate the built-in signer, you must pass the `--cluster-signing-cert-file` and `--cluster-signing-key-file` flags.
 
-If you're creating a new cluster, you can use a kubeadm [configuration file](https://godoc.org/k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1beta2):
+If you're creating a new cluster, you can use a kubeadm [configuration file](/docs/reference/config-api/kubeadm-config.v1beta2/):
```
Comment (sftim) [inline content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-certs.md:164]: Aside: the reader might be none the wiser as to where to put this YAML on their filesystem to have it used. Or maybe they pass it on stdin?

We could clarify that (could be a separate PR).

File: content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md [hunk 3]
```diff
@@ -122,8 +122,8 @@ The default configuration can be printed out using the
 If your configuration is not using the latest version it is **recommended** that you migrate using
 the [kubeadm config migrate](/docs/reference/setup-tools/kubeadm/kubeadm-config/) command.
 
-For more information on the fields and usage of the configuration you can navigate to our API reference
-page and pick a version from [the list](https://pkg.go.dev/k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm#section-directories).
+For more information on the fields and usage of the configuration you can navigate to our
+[API reference page](/docs/reference/config-api/kubeadm-config.v1beta2/).
```
Comment (neolit123) [inline content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md:126]: is there a way to navigate to a page that lists all kubeadm APIs?

that is what the godoc link is doing:
https://pkg.go.dev/k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm#section-directorie

so given this is for the main branch we should link to v1beta2.
and for the dev-1.22 branch we should link to v1beta3?

File: content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-certs.md (hunk 2 above)
Comment (neolit123) [reply content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-certs.md:164]: the linked page already includes details (under the "Basics" section) that a file can be passed to kubeadm with "--config".

**Comment by jihoon-seo** [comment]
/unhold
The config reference PR #28871 was merged.

File: content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md (hunk 3 above)
Comment (kbhawkey) [reply content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md:126]: @tengqm , Does it make sense to link to configuration API index page?
https://kubernetes.io/docs/reference/config-api/

Comment (tengqm) [reply content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md:126]: @neolit123 @kbhawkey  I'm not sure we have such a use case for linking to both versions. In 1.21 (main), we point users to v1beta2; while in 1.22, we point users to v1beta3. If there IS a strong use case for users to browse through the different versions, we can change the link to the config-api section. That also means we may need a dedicated subsection for kubeadm configs. Make sense?

Comment (neolit123) [reply content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md:126]: pointing to v1beta3 only seems mostly fine, but we have to keep in mind that it would take 1 year for users to migrate to the new version. having the reference for the old v1beta2 here makes sense.

Comment (sftim) [reply content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md:126]: :+1: to 
> having the reference for the old v1beta2 here makes sense.

based on justification above.
(I'm happy to see this change merge, that'd be a separate improvement).

**Comment by kbhawkey** [comment]
Thanks @tengqm .
/approve
